R-packages/modeltools/vignettes/estimate-deriv.Rmd

```diff
@@ -0,0 +1,139 @@
+---
+title: 3. Estimate derivatives of signals 
+description: Estimate derivatives of signal values, using various methodologies.
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{3. Estimate derivatives of signals}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(message = FALSE, warning = FALSE)
+knitr::opts_chunk$set(cache = TRUE, autodep = TRUE, cache.comments = TRUE)
+```
+
+TODO, blah load data
+
+```{r}
+library(covidcast)
+
+start_day <- "2020-06-01"
+end_day <- "2020-11-15"
+geo_values <- c("ca", "fl", "ny", "tx")
+
+case_rates <- suppressMessages(
+  covidcast_signal(data_source = "usa-facts", 
+                   signal = "confirmed_7dav_incidence_prop",
+                   start_day = start_day, end_day = end_day, 
+                   geo_type = "state", geo_values = geo_values))
+
+summary(case_rates)
+```
+
+## Estimating derivatives via linear regression
+
+TODO, blah try linear regression
+
+```{r}
+library(modeltools)
+library(dplyr)
+
+case_rates <- estimate_deriv(case_rates, method = "lin", n = 14)
+
+case_rates %>% 
+  arrange(geo_value) %>% 
+  select(geo_value, time_value, value, deriv) 
+```
+
+TODO, blah plot it
+
+```{r, fig.width = 8, fig.height = 4}
+library(ggplot2)
+library(gridExtra)
+
+state = "fl"
+threshold = 0.25
+
+p1 <- ggplot(case_rates %>% filter(geo_value == state),
+             aes(x = time_value, y = value)) +
+  geom_line() + 
+  geom_point(data = case_rates %>% 
+               filter(geo_value == state, deriv >= threshold),
+             aes(x = time_value, y = value), color = "red") + 
+  labs(x = "Date", y = "Cases per 100,00 people") 
+
+p2 <- ggplot(case_rates %>% filter(geo_value == state),
+             aes(x = time_value, y = deriv)) +
+  geom_line() + 
+  geom_hline(yintercept = threshold, linetype = 2) +
+  labs(x = "Date", y = "Derivative (linear regression)") 
+
+grid.arrange(p1, p2, nrow = 1)
+```
+
+## Estimating derivatives via smoothing spline
+
+TODO, blah try smoothing splines two ways, plot them
+
+```{r, fig.width = 8, fig.height = 4}
+case_rates <- estimate_deriv(case_rates, method = "ss", n = 28,
+                             new_col = "deriv_ss1", df = 8)
+
+case_rates <- estimate_deriv(case_rates, method = "ss", n = 28,
+                             new_col = "deriv_ss2", cv = TRUE)
```

Comment:
```suggestion
                             col_name = "deriv_ss2", cv = TRUE)

Reply:
Good catch, you spotted this before I pushed my latest commits (which resolve these inconsistencies).